applications/coreutils/dirname.cpp

```diff
@@ -0,0 +1,55 @@
+#include <libsystem/cmdline/CMDLine.h>
+#include <libsystem/io/Stream.h>
+#include <libsystem/io/Path.h>
+
+static bool zero = false;
+
+static const char *usages[] = {
+    "NAME...",
+    "[OPTION] NAME...",
+    nullptr,
+};
+
+static CommandLineOption options[] = {
+    COMMANDLINE_OPT_HELP,
+
+    COMMANDLINE_OPT_BOOL("zero", 'z', zero,
+                         "end each output line with NUL, not newline",
+                         COMMANDLINE_NO_CALLBACK),
+    COMMANDLINE_OPT_END};
+
+static CommandLine cmdline = CMDLINE(
+    usages,
+    options,
+    "Output each NAME with its last non-slash component and trailing slashes removed;",
+    "if NAME contains no  /'s,  output  '.'  (meaning  the  current directory).");
+
+const char *dirname(Path *path)
```

Comment:
You should return a simple `char *`, and then you should `free()` it

Reply:
Oh okay, I'll make all the changes 😄